backend/app/modules/computation/Cluster.scala

```diff
@@ -0,0 +1,94 @@
+package modules.computation
+
+import scala.annotation.tailrec
+import scala.collection.mutable
+
+/**
+  * Represents a cluster of pixels.
+  *
+  * This is mostly a utility class to manipulate pixels.
+  *
+  * @param pixels Pixels composing the cluster.
+  */
+case class Cluster(var pixels: List[Pixel]) {
+
+  /** Shortcut for cluster size */
+  var size: Int = pixels.size.intValue()
+
+  /** Cluster barycentre */
+  var x: Double = pixels.map(_.x).foldLeft(0)(_ + _) / size.toFloat
```

Comment:
I have issues with this one where I couldn't get rid of `could not find implicit value for parameter num: Numeric[Integer]`.